oauth2/handler.go

```diff
@@ -250,6 +250,9 @@ func (h *Handler) AuthHandler(w http.ResponseWriter, r *http.Request, _ httprout
 		h.writeAuthorizeError(w, authorizeRequest, errors.Wrapf(fosite.ErrServerError, "Could not store session cookie: %s", err))
 		return
 	}
+	
+	// add key id to header
+	session.Headers.Add("kid", "public")
```

Comment:
@arekkas I know you had mentioned putting this here https://github.com/ory/hydra/blob/master/oauth2/session.go#L21 but fosite appears to not be passing the session headers though entirely. 

Even if the token headers are populated in the session, https://github.com/ory/hydra/blob/master/oauth2/handler.go#L169 appears to strip them. If we want a different solve we will likely need to move into fosite.